workflow.rst

```diff
@@ -508,6 +508,12 @@ missing a title::
 .. versionadded:: 5.1
 
     The optional second argument of ``setBlocked()`` was introduced in Symfony 5.1.
+    
+.. note::
+
+    When using guard listeners which imply intensive workloads (CPU, Database or longer-running code blocks), if you only want them to be fired when strictly necesary (only when ``Workflow::can()`` or ``Workflow::apply()`` is executed), be sure to disable ``Workflow::DISABLE_ANNOUNCE_EVENT`` as indicated in :ref:`Choosing which Events to Dispatch <workflow-chosing-events-to-dispatch>`
```

Comment:
> When using guard listeners which imply intensive workloads

hum by default guard imply only what you call inside your guard listeners/subscriber methods, isn't it?
if your custom code is heavy/long/etc, I think it is your responsibility to make it cacheable in some way for current request process flow to avoid this `intensive workloads`

---

also for documentation, lines chars number must be truncated, see for exemple the next p `If you prefer to control...`

Reply:
Well there are some cases when you cannot do some caching, and in those you want to make as few calls as possible, thus eliminating the need for `getEnabledTransitions()` . This can save quite a lot of request time.